src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_CAPACITY = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+    private int capacity;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+        capacity = DEFAULT_CAPACITY;
+    }
+
+    static class Node<K,V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private void resize() {
+        size = 0;
+        capacity *= 2;
+        threshold *= 2;
+        Node<K, V>[] oldTable = table;
+        Node<K, V>[] newTable = new Node[capacity];
+        table = newTable;
+        for (Node<K, V> node: oldTable) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
+    }
+
+    private int hashValue(K key) {
+        return key == null ? 0 : key.hashCode() * 11;
+    }
+
+    private int getIndex(int hashValue) {
+        return hashValue > 0 ? hashValue % capacity : -1 * hashValue % capacity;
+    }
```

Comment:
let's use Math.abs to always get positive index here and remove ternary operator